megatron/learning_rates.py

```diff
@@ -64,6 +64,19 @@ def get_lr(self):
         """Learning rate decay functions from:
               https://openreview.net/pdf?id=BJYwwY9ll pg. 4"""
 
+
+        if self.decay_style == 'inverse_sqrt':
+            if self.warmup_steps > 0 and self.num_steps <= self.warmup_steps:
+                if self.warmup_style == 'linear':
+                    return self.max_lr * self.num_steps / (self.warmup_steps * self.warmup_steps**0.5)
+                elif self.warmup_style == 'constant':
+                    return self.max_lr / self.warmup_steps**0.5
+                else:
+                    raise ValueError('Unknown warmup style: {}'.format(
+                        self.warmup_style))
+                
+            return self.max_lr / (max(self.num_steps, 1))**0.5
```

Comment:
Actually I think I was wrong, but this should be something like 
```
return self.max_lr / (max(self.num_steps, max(self.warmup_steps, 1)))**0.5
```
But I'd argue that this should still be done after :D